src/sentry/static/sentry/app/utils/queryString.tsx

```diff
@@ -51,26 +51,26 @@ export function appendTagCondition(
   return currentQuery;
 }
 
-export function decodeScalar(
-  value: string[] | string | undefined | null
-): string | undefined {
+// This function has multiple signatures to help with typing in callers.
+export function decodeScalar(value: QueryValue): string | undefined;
```

Comment:
Yeah these overloads make the types nicer.

Reply:
Using an overload only came to mind after struggling with the tests for a few minutes, and then seeing the fixes causing other failures. :blush: